### PR TITLE
refactor(meta): decouple steps of stream job preparation (part 1)

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Detailedly explained in #7420. This PR mainly focuses on the internal table handling.

- Extract the visiting logic into a single function of `visit_internal_table`.
- Move the logic of filling IDs before constructing the **Fragment** Graph, so that once the graph is built, it's complete.
- Remove a bunch of (mutable) fields in the `Context`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- #7420

### CODE REVIEW

**Review by yezizp2012 — APPROVED** [review]
LGTM!

**Review by chenzl25 — APPROVED** [review]
LGTM!

File: src/meta/src/stream/stream_graph.rs [hunk 1]
```diff
@@ -524,31 +534,10 @@ impl StreamGraphBuilder {
     /// ids if it is a `ChainNode`.
     fn build_inner(
         &self,
-        ctx: &mut CreateStreamingJobContext,
         stream_node: &StreamNode,
-        fragment_id: GlobalFragmentId,
         upstream_actor_id: &mut HashMap<u64, OrderedActorLink>,
         upstream_fragment_id: &mut HashMap<u64, GlobalFragmentId>,
     ) -> MetaResult<StreamNode> {
```
Comment (xx01cyx) [inline src/meta/src/stream/stream_graph.rs:540]: The comments of this method can by the way be refactored as well.

**Review by xx01cyx — APPROVED** [review]
LGTM!

File: src/meta/src/stream/stream_graph.rs (hunk 1 above)
Comment (BugenZhao) [reply src/meta/src/stream/stream_graph.rs:540]: This doc is not updated when the logic of internal tables is mixed into this function. So after we move it out, the current doc exactly reflects what it does and there's no need to update it. 😂

**Pull request merged**